### DISCUSSION TITLE
bug: decodeEventLog returns an array without topics

### DISCUSSION THREAD

**Discussion by jagnani73** [wevm/viem#1821] [Idea / Feature Request]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

^1.16.6

### Current Behavior

If the ABI event has any input without a name, the following issues occur in decodeEventLog:
1. The args are returned in an array instead of an object
2. Only the data is parsed and returned, the topics are not returned

### Expected Behavior

1. The inputs with a name and the topics should be returned in a structured object
2. The inputs without a name should either have a pseudo key name or be clubbed in an array separate from the originally returned args

### Steps To Reproduce

1. Use the following data as arguments for the decodeLogEvent (refer [this](https://etherscan.io/tx/0x58779b2c975844399aa5d4e1a9c7f1a035098e1ca32b6c2755ee6fa6194b75f1#eventlog) transaction)
	a. abi: https://etherscan.io/address/0x5c69bee701ef814a2b6a3edd4b1652cb9cc5aa6f#code:~:text=Contract%20ABI,Export%20ABI
	b. data: `0x0000000000000000000000006cf029fe9c585f666255ff021c5114b6225a7bb0000000000000000000000000000000000000000000000000000000000004ac9c`
	c. topics: `['0x0d3648bd0f6ba80134a33ba9275ac585d9d315f0ad8355cddefde31afa28d0e9', '0x0000000000000000000000008f6e363022a5612003d62f2d29ac5ea6b37bb1e0', '0x000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2']`
	d. eventName: `PairCreated`

### Anything else?

The expected result is achieved by modifying the ABI and putting a name for the unnamed input

**Comment by jxom** [comment]
Args are decoded into an array as one of the arguments are unnamed – which is intentional behaviour. Could possibly look into its index being the name to preserve the object.

Going to close this as it isn't really a bug, but will convert into a discussion to continue the conversation.

**Reply by jxom** [reply]
Though, it is weird that the indexed arguments aren't being decoded? Can take a look at that.

**Reply by jagnani73** [reply]
Thanks
